src/test/scala/ru/smslv/akka/dns/AsyncDnsResolverSpec.scala

```diff
@@ -99,4 +101,43 @@ class AsyncDnsResolverSpec extends AkkaSpec(
       answer.ipv6 should be(empty)
     }
   }
+
+  "Resolver functions" should {
+    "parse a resolv.conf file successfully" in {
+      val resolvConf = Files.createTempFile("resolve", "conf").toFile
+      resolvConf.deleteOnExit()
+      Files.write(
+        resolvConf.toPath,
+        """# This file is automatically generated.
+          |#
+          |search lan
+          |nameserver 10.0.1.1
+          |nameserver 2001:44b8:3127:3500::1
+          |
+        """.stripMargin.getBytes
+      )
+      AsyncDnsResolver.parseSystemNameServers(resolvConf) should be(Some(List(
+        AsyncDnsResolver.parseNameserverAddress("10.0.1.1"),
+        AsyncDnsResolver.parseNameserverAddress("2001:44b8:3127:3500::1")
+      )))
+    }
+
+    "return an empty list when a resolv.conf file has no nameservers" in {
+      val resolvConf = Files.createTempFile("resolve", "conf").toFile
+      resolvConf.deleteOnExit()
+      Files.write(
+        resolvConf.toPath,
+        """# This file is automatically generated.
+          |#
+          |search lan
+        """.stripMargin.getBytes
+      )
+      AsyncDnsResolver.parseSystemNameServers(resolvConf) should be(Some(List.empty))
+    }
+
+    "return nothing when there is no resolv.conf" in {
+      val resolvConf = new File("something-somewhere")
+      AsyncDnsResolver.parseSystemNameServers(resolvConf) should be(None)
+    }
+  }
```

Comment:
So you get no feedback if you have a typo in the resolve.conf? (if it doesn't match the regex)